sway-core/src/asm_generation/declaration/reassignment.rs

```diff
@@ -22,8 +23,8 @@ pub(crate) fn convert_reassignment_to_asm(
     // 2. move the return register of the RHS into the register in the namespace
 
     let mut buf = vec![];
-    let mut warnings = vec![];
-    let mut errors = vec![];
+    let mut warnings = IndexSet::new();
+    let mut errors = IndexSet::new();
```

Comment:
This is definitely a nice library, awesome!

Now that this pattern is getting altered, I think it'd be a good chance to introduce an abstraction over errors and warnings. 

I think if we are using a third party collection, it would be wise to wrap it in an interface. That interface could enable us to have a new API for initializing these collections. Something like:
```suggestion
    let (mut warnings, mut errors) = ErrorSet::init();
```

Then, elsewhere, we could have:
```rust
struct ErrorSet {
   warnings: IndexSet<CompileWarning>,
   errors: IndexSet<CompileWarning>,
}
```

This also unblocks simplifying the interface in `CompileResult`, but that'd be out of scope to include here.

The motivation for this is that we would not need to have another high-touch PR if we want to alter the collection or third party library we are using.